crates/bevy_ui/src/widget/slider.rs

```diff
@@ -0,0 +1,171 @@
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::prelude::Component;
+use bevy_ecs::query::With;
+use bevy_ecs::reflect::ReflectComponent;
+use bevy_ecs::schedule::SystemLabel;
+use bevy_ecs::system::{Query, Res};
+use bevy_hierarchy::Children;
+use bevy_input::prelude::MouseButton;
+use bevy_input::touch::Touches;
+use bevy_input::Input;
+use bevy_reflect::std_traits::ReflectDefault;
+use bevy_reflect::Reflect;
+use thiserror::Error;
+
+use crate::{Interaction, Node, RelativeCursorPosition, Style, Val};
+
+/// A component describing the slider-specific value, such as max and min values and step
+#[derive(Component, Debug, Clone, Copy, Reflect)]
+#[reflect(Component, Default)]
+pub struct Slider {
+    min: f32,
+    max: f32,
+    step: f32,
+    value: f32,
```

Comment:
Can we remove `min` and `max` and instead keep value in the range `0.0..=1.0`? That reduces the API surface and I think also allows us to get rid of the `step` field?

This is my plan for #6517 as #6620 was deemed not desired.

Reply:
Removed `min` and `max`, but I don't see how that makes the `step` redundant?